src/modules/Host.rb

```diff
@@ -292,10 +291,11 @@ def StaticIPs
     # make sure /etc/hosts resolves it to our, bnc#664929
     def ResolveHostnameToStaticIPs
       static_ips = StaticIPs()
-      if Ops.greater_than(Builtins.size(static_ips), 0)
-        fqhostname = Hostname.MergeFQ(DNS.hostname, DNS.domain)
-        Update(fqhostname, fqhostname, static_ips)
-      end
+      return if static_ips.empty?
+
+      fqhostname = Hostname.MergeFQ(DNS.hostname, DNS.domain)
+
+      static_ips.reject(&:empty?).each { |sip| Update(fqhostname, fqhostname, sip) }
```

Comment:
`StaticIPs` is defined in this file and this method is the only user. My conclusion is to instead fix that method not to return invalid data in the first place.